src/utils/wallet.js

```diff
@@ -211,6 +214,34 @@ const wallet = {
     }
   },
 
+  /**
+   * The wallet needs each token metadata to show information correctly
+   * So we fetch the tokens metadata and store on redux
+   *
+   * @param {Array} tokens Array of token uids
+   * @param {String} network Network name
+   *
+   * @memberof Wallet
+   * @inner
+   **/
+  async fetchTokensMetadata(tokens, network) {
+    const metadatas = {};
+
+    for (const token of tokens) {
+      try {
+        const response = await metadataApi.getDag(token, network);
+        if (response.data && token in response.data) {
```

Comment:
What does it mean when we do not receive a token in the response? Can an unexpected error cause it somehow?

Reply:
I've refactored this part of code to have an error handling and auto retry. Can you see if it's good? https://github.com/HathorNetwork/hathor-wallet/pull/228/commits/9e150da7e7f5923df00911378430609f9345f9aa

Reply:
Well done. Thanks!